R/bundle.R

```diff
@@ -164,7 +164,12 @@ bundleApp <- function(appName, appDir, appFiles, appPrimaryDoc, assetTypeName,
                       contentCategory) {
 
   # infer the mode of the application from its layout
-  appMode <- inferAppMode(appDir, appPrimaryDoc, appFiles)
+  # unless we're an API, in which case, we're API mode.
+  if (contentCategory == "api") {
```

Comment:
(can't recall if contentCategory can be NULL in this case, so if you can convince yourself it will always be a character vector of length one then what you've got is fine!)